src/Orckestra.Composer.SearchQuery/Context/SearchQueryContext.cs

```diff
@@ -13,28 +20,100 @@ public class SearchQueryContext : ISearchQueryContext
         protected IComposerContext ComposerContext { get; private set; }
         protected ISearchQueryViewService SearchQueryViewService { get; private set; }
         protected ISearchViewService SearchViewService { get; private set; }
+        protected IInventoryLocationProvider InventoryLocationProvider { get; }
+        protected ISearchUrlProvider SearchUrlProvider { get; }
 
-        protected SearchQueryViewModel ViewModel { get; set; }
+        private SearchQueryViewModel _viewModel { get; set; }
+        private SearchQueryViewModel _topResultsViewModel { get; set; }
+        protected HttpRequestBase Request { get; private set; }
+        public virtual string SortBy => Request[SearchRequestParams.SortBy];
+        public virtual string SortDirection => Request[SearchRequestParams.SortDirection] ?? SearchRequestParams.DefaultSortDirection;
 
-        public SearchQueryContext(IComposerContext composerContext, ISearchQueryViewService searchQueryViewService, ISearchViewService searchViewService)
+        public virtual int CurrentPage
+        {
+            get
+            {
+                return int.TryParse(Request[SearchRequestParams.Page], out int page) && page > 0 ? page : 1;
+            }
+        }
+
+        public SearchQueryContext(IComposerContext composerContext,
+            ISearchQueryViewService searchQueryViewService,
+            ISearchViewService searchViewService,
+            IInventoryLocationProvider inventoryLocationProvider,
+            ISearchUrlProvider searchUrlProvider,
+            HttpRequestBase request)
         {
             ComposerContext = composerContext ?? throw new ArgumentNullException(nameof(composerContext));
             SearchQueryViewService = searchQueryViewService ?? throw new ArgumentNullException(nameof(searchQueryViewService));
             SearchViewService = searchViewService ?? throw new ArgumentNullException(nameof(searchViewService));
+            InventoryLocationProvider = inventoryLocationProvider ?? throw new ArgumentNullException(nameof(inventoryLocationProvider));
+            Request = request ?? throw new ArgumentNullException(nameof(request));
+            SearchUrlProvider = searchUrlProvider ?? throw new ArgumentNullException(nameof(searchUrlProvider));
+        }
+
+        public async Task<SearchQueryViewModel> GetSearchQueryViewModelAsync(SearchQueryType queryType, string queryName)
```

Comment:
It little bit strange to get parameters and continue to receive parameters from querystring

Reply:
It is because Context which can bu used in several Razor function on same page. To avoid duplicate code in Razors, Context reads these parameters from Request.